builds/e2e/nested-longhaul.yaml

```diff
@@ -21,40 +21,96 @@ resources:
     branch: 'release/1.2'
 
 stages:
-- stage: SetupVMs
+- stage: SetupAgents 
+  jobs:
+  - template: templates/lock-test-agents.yaml
+    parameters:
+      upstream.protocol: 'amqp'
+  - template: templates/lock-test-agents.yaml
+    parameters:
+      upstream.protocol: 'mqtt'
+
+  # Get the ip address of the L3 agent. Needed for test modules on the middle level L4 agent.
+  - job: Get_L3_IP_mqtt
+    dependsOn:
+      - Lock_Nested_Agents_mqtt
+    displayName: Get ip address of L3 mqtt agent
+    timeoutInMinutes: 180
+    pool:
+     name: $(pool.name)
+     demands:
+       - agent-group -equals $(agent.group)
+       - Agent.OS -equals Linux
+       - Agent.OSArchitecture -equals X64
+       - status -equals unlocked_$(Build.BuildId)_L3_mqtt
+    steps:
+      - script: |
+          L3_IP_ADDRESS_MQTT=$(ip route get 8.8.8.8 | sed -n '/src/{s/.*src *\([^ ]*\).*/\1/p;q}')
+          echo "Ip adress of L3 agent: $L3_IP_ADDRESS_MQTT"
+          echo "##vso[task.setvariable variable=l3IpAddressMqtt;isOutput=true]${L3_IP_ADDRESS_MQTT}"
+        displayName: Get ip address of L3 mqtt agent
+        name: get_l3_ip_mqtt
+
+  # Generate tracking id
+  - job: Generate_Tracking_Id
+    dependsOn:
+      - Lock_Nested_Agents_mqtt
+    displayName: Generate test tracking id
+    timeoutInMinutes: 180
+    pool:
+     name: $(pool.name)
+     demands:
+       - agent-group -equals $(agent.group)
+       - Agent.OS -equals Linux
+       - Agent.OSArchitecture -equals X64
+    steps:
+      - script: |
+          tracking_id=$(cat /proc/sys/kernel/random/uuid)
+          echo "Test tracking id: $tracking_id"
+          echo "##vso[task.setvariable variable=trackingId;isOutput=true]${tracking_id}"
+        displayName: Generate Tracking Id
+        name: generate_tracking_id
```

Comment:
The above stage will lock all agents, then set devops variables for a generated tracking id and L3 ip address.
- Tracking Id: Needs to be generated and shared between levels so that middle level and lower level test modules can use the same tracking id. This is necessary because of the TRC architecture.
- L3 ip address: Needed so that when it comes time to deploy the L4 level, we can inject the ip address of the L3 level. This is needed so that test modules can talk to the TRC on the L3 node.